### ISSUE TITLE
Figure out how to handle private channels

### ISSUE THREAD

**Issue by wolfv** [TheSnakePit/quetz#81]
I think while we have private channels, we need to implement how we handle them properly. I think right now every user can see the contents of private channels, for example.

**Comment by adriendelsalle** [comment]
Related to #54 ?

**Comment by wolfv** [comment]
yes but this was also about the `/api/channels` for example. And `/api/channels/channelX/packages`.
I think they should return different things depending on the permission level.

I need to finish this one tricky join today so that this hides private channels that User is not a member of.

**Comment by adriendelsalle** [comment]
Nice!
I agree, this what I called a user *view*. It apply to the repodata.json but also on any other requests to the api.

**Comment by wolfv** [comment]
@adriendelsalle in this PR I have added search, that only returns packages that the logged in user has access to: https://github.com/TheSnakePit/quetz/pull/82

I want to add tests next week so that we can check that this behaves as intended.

**Comment by adriendelsalle** [comment]
#105
